assets/js/base.js

```diff
@@ -209,5 +217,127 @@ window.Herman = (function base(Herman, $) {
+                length: p[1],
+              })
+            );
+            // @@@ Temporary solution until fieldname bug is fixed:
+            // https://github.com/olivernn/lunr.js/issues/320
+            const field = fieldName === 'title' ? 'title' : 'contents';
+            toMark[field] = toMark[field].concat(pos);
+          });
+        });
+        if (!(toMark.title.length || toMark.contents.length)) {
+          return;
+        }
+        const obj = {
+          url: `/${res.ref}`,
+          title: doc.title,
+          contents: toMark.contents.length ? doc.contents : '',
+        };
+        // Render search result template
+        const el = $(window.nunjucks.render('search_result.j2', obj));
+        if (toMark.title.length) {
+          // Highlight matches in `title` field
+          el.find(`[data-result-field="title"]`).markRanges(toMark.title);
+        }
+        if (toMark.contents.length) {
+          const textEl = el.find(`[data-result-field="contents"]`);
+          toMark.contents.sort((a, b) => a.start - b.start);
+          // Highlight first 5 matches in `contents` field
+          textEl.markRanges(toMark.contents.slice(0, 5), {
+            done: () => {
+              // Truncate text not within 15 words of a match
+              textEl.get(0).childNodes.forEach(node => {
+                const hasPrev = node.previousSibling !== null;
+                const hasNext = node.nextSibling !== null;
+                const isText = node.nodeName === '#text';
+                if (isText) {
+                  const text = node.nodeValue.split(' ');
+                  if (hasPrev && hasNext) {
+                    if (text.length > 30) {
+                      text.splice(15, text.length - 30, '…');
+                    }
+                  } else if (hasNext) {
+                    if (text.length > 15) {
+                      text.splice(0, text.length - 15, '…');
+                    }
+                  } else if (hasPrev) {
+                    if (text.length > 15) {
+                      text.splice(15, text.length - 15, '…');
+                    }
+                  }
+                  node.nodeValue = text.join(' ');
+                }
+              });
+            },
+          });
+        }
+        // Add search result template to set of results
+        matches = matches.add(el);
+      }
+    }
+    const tpl = $(
+      window.nunjucks.render('search_results.j2', {
+        term: val,
+        count: matches.length,
+      })
+    );
+    tpl.filter('.js-search-results').html(matches);
+    $('[data-sassdoc-region="main"]').html(tpl);
+  };
+
+  const doSearch = function(data, val) {
+    // Grab doc store from data
+    Herman.searchStore = data.store;
+    // Initialize Lunr index from precompiled data
+    const idx = window.lunr.Index.load(data.idx);
+    const results = idx.search(val);
+    showResults(results, val);
+  };
+
+  Herman.getSearchData = function getSearchData() {
```

Comment:
Why isn't this an anonymous function assigned to `Herman.getSearchData`?

Reply:
Good call; no reason.